etcdctl/ctlv3/command/perf_validate.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2017 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package command
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"math"
+	"math/rand"
+	"sync"
+	"time"
+
+	v3 "github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/pkg/report"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/time/rate"
+	"gopkg.in/cheggaaa/pb.v1"
+)
+
+var (
+	validPerfLoad string
+)
+
+type validatePerfCfg struct {
+	limit    int
+	clients  int
+	duration int
+}
+
+var validatePerfCfgMap = map[string]validatePerfCfg{
+	// TODO: support read limit
+	"s": validatePerfCfg{
+		limit:    150,
+		clients:  50,
+		duration: 60,
+	},
+	"m": validatePerfCfg{
+		limit:    1000,
+		clients:  200,
+		duration: 60,
+	},
+	"l": validatePerfCfg{
+		limit:    8000,
+		clients:  500,
+		duration: 60,
+	},
+	"xl": validatePerfCfg{
+		limit:    15000,
+		clients:  1000,
+		duration: 60,
+	},
+}
+
+// NewValidatePerfCommand returns the cobra command for "validate-perf".
+func NewValidatePerfCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "validate-perf [options]",
```

Comment:
If it's going to be `check-perf`, then `check perf` might be better. There's probably a few aspects of the cluster worth checking aside from only performance (e.g., check if running latest version, check if hashes match, ...) so `check` could cover all those as subcommands.